c/meterpreter/source/extensions/stdapi/server/net/socket/tcp_server.c

```diff
@@ -301,11 +301,6 @@ DWORD request_net_tcp_server_channel_open(Remote * remote, Packet * packet)
 		ctx->remote = remote;
 
 		localPort = (USHORT)(packet_get_tlv_value_uint(packet, TLV_TYPE_LOCAL_PORT) & 0xFFFF);
-		if (!localPort)

```

Comment:
Removing this makes this implementation consistent with the Python and mettle implementations where when binding to port 0, it will be selected by the OS and (now with these changes) metasploit will be informed as to the selection.